docs/how-to-build-with-ddn/with-mongodb.mdx

```diff
@@ -0,0 +1,399 @@
+
+```sh title="Next, use the CLI to introspect your MongoDB database:"
+ddn connector introspect my_mongo
+```
+
+After running this, you should see a representation of your collection's schema in
+`app/connector/my_mongo/schema/users.json`; you can view this using `cat` or open the file in your editor.
+
+For each collection in your database, the MongoDB connector will generate a separate JSON file representing it.
+
+### Step 7. Add your model
+
+```sh title="Now, track the collection from your MongoDB database as a model:"
+ddn models add my_mongo users
+```
+
+Open the `app/metadata` directory and you'll find a newly-generated file: `Users.hml`. The Hasura engine will use this
+Hasura Metadata Language file to bring the `users` table from MongoDB into your API as a
+[model](/supergraph-modeling/models.mdx).
+
+### Step 8. Create a new build
+
+```sh title="To create a local build, run:"
+ddn supergraph build local
+```
+
+The build is stored as a set of JSON files in `engine/build`.
+
+### Step 9. Start your local services
+
+```sh title="Start your local Hasura DDN Engine and MongoDB connector:"
+ddn run docker-start
+```
+
+Your terminal will be taken over by logs for the different services.
+
+### Step 10. Run your first query
+
+```sh title="In a new terminal tab, open your local console:"
+ddn console --local
+```
+
+```graphql title="In the GraphiQL explorer of the console, write this query:"
+query {
+  users {
+    userId
+    name
+    age
+  }
+}
+```
+
+```json title="You'll get the following response:"
+{
+  "data": {
+    "users": [
+      {
+        "userId": 1,
+        "name": "Alice",
+        "age": 25
+      },
+      {
+        "userId": 2,
+        "name": "Bob",
+        "age": 30
+      },
+      {
+        "userId": 3,
+        "name": "Charlie",
+        "age": 35
+      }
+    ]
+  }
+}
+```
+
+### Step 11. Iterate on your MongoDB schema
+
+```sh title="Let's add a new collection for posts:"
+
+docker exec -it mongodb mongosh --eval "
+db = db.getSiblingDB('my_database');
+db.posts.insertMany([
+  { user_id: 1, title: 'My First Post', content: 'This is Alice\'s first post.' },
+  { user_id: 1, title: 'Another Post', content: 'Alice writes again!' },
+  { user_id: 2, title: 'Bob\'s Post', content: 'Bob shares his thoughts.' },
+  { user_id: 3, title: 'Hello World', content: 'Charlie joins the conversation.' }
+]);
+"
+```
+
+### Step 12. Refresh your metadata and rebuild your project
+
+:::tip
+
+The following steps are necessary each time you make changes to your **source** schema. This includes, adding,
+modifying, or dropping collections.
+
+:::
+
+#### Step 12.1. Re-introspect your data source
+
+```sh title="Run the introspection command again:"
+ddn connector introspect my_mongo
+```
+
+In `app/connector/my_mongo/configuration.json`, you'll see schema updated to include operations for the `posts`
+collection. You'll also see a `posts.json` file in the `schema` directory.
+
+#### Step 12.2. Update your metadata
```

Comment:
Same deal, I think we need a connector restart between steps 12.1 and 12.2 until the "print schema" capability is implemented

Reply:
Is it possible there's some caching of a docker build going on? Despite adding this step in, I'm still having to manually delete the image and then rerun `ddn run docker-start` to get my changes to appear in the connector.

Reply:
Same. I get this error: 

<img width="1164" alt="Screenshot 2024-12-04 at 13 10 21" src="https://github.com/user-attachments/assets/c10e6669-1dd4-4ce6-8a38-99c8b6aa80c7">